### PR TITLE
docs(errors): add Error messaging page and section

### LINKED ISSUE #513: [Docs/Core] Audit, standardize and document all Error messaging in app
- https://github.com/reactioncommerce/reaction/pull/3538/files#diff-9ef1c7def4fdab310138484898aa6368R100 from @aaronjudd 

Document what errors we have now for the actual error messaging.

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #513

I added information from https://github.com/reactioncommerce/reaction/pull/3195 for
```access-denied
not-implemented
invalid-parameter
not-configured
server-error
not-found
``` 
with some background information on `Meteor.Error()` from the Meteor guide.

Test: https://docs.reactioncommerce.com/reaction-docs/513-machikoyasuda-error-messaging/error-handling-guide

### CODE REVIEW

File: developer/error-handling-guide.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Error handling guide
```
Comment (machikoyasuda) [inline developer/error-handling-guide.md]: Do you think this is a good title/filename for this topic? Wasn't sure if `Error messaging guide` would be better.

Comment (machikoyasuda) [reply developer/error-handling-guide.md]: Alternatively, all this info could go in the Code Style Guide at the end: https://docs.reactioncommerce.com/reaction-docs/master/styleguide#copy-conventions

File: developer/error-handling-guide.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+| `not-configured`    | "Open Exchange Rates not configured. Configure for current rates."  |
+| `not-found`         | "User not found"                                                    |
+| `not-implemented`   | "Multiple shipping providers is currently not implemented"          |
+| `server-error`      | "Error adding to cart"                                              |
+
+### Error message guidelines
+
+The above list is not exhaustive throughout the whole app. For other error scenarios, use the guidelines below:
+
+- Do not use HTTP status codes, like `404` or `500`. Use the error messages `invalid-parameter` and `internal-server-error` instead respectively.
```
Comment (zenweasel) [inline developer/error-handling-guide.md]: We say `server-error` above but then you say `internal-server-error` here. Should probably be consistent?

File: developer/error-handling-guide.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- Always start the reason with a capital letter.
+- Do not use exclamation marks in the reason.
```
Comment (zenweasel) [inline developer/error-handling-guide.md]: lol, yeah. Don't do that

File: developer/error-handling-guide.md (hunk 1 above)
Comment (zenweasel) [reply developer/error-handling-guide.md]: I think so. There's certainly a ton more you could say about "Error Handling"but this is a good start.

**Pull request merged**